omnigibson/prims/rigid_prim.py

```diff
@@ -453,15 +398,11 @@ def mass(self):
         Returns:
             float: mass of the rigid body in kg.
         """
-        raw_usd_mass = self._mass_api.GetMassAttr().Get()
-        # If our raw_usd_mass isn't specified, we check dynamic control if possible (sim is playing),
-        # otherwise we fallback to analytical computation of volume * density
-        if raw_usd_mass != 0:
-            mass = raw_usd_mass
-        elif self.dc_is_accessible:
-            mass = self.rigid_body_properties.mass
-        else:
-            mass = self.volume * self.density
+        mass = self._rigid_prim_view.get_masses()[0]
+
+        # Fallback to analytical computation of volume * density
+        if mass == 0:
+            return self.volume * self.density
```

Comment:
is this how omni computes mass? If not, should we enforce consistency by immediately setting the mass value here?

Reply:
I believe so, see https://docs.omniverse.nvidia.com/extensions/latest/ext_physics/rigid-bodies.html#mass-properties

Reply:
Agreed that we should do something about this, but let's do it later. Filed #351